src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayViewController.swift

```diff
@@ -0,0 +1,174 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import Combine
+
+class DiaryDayViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+
+	// MARK: - Init
+
+	init(
+		viewModel: DiaryDayViewModel,
+		onAddEntryCellTap: @escaping (DiaryDay, DiaryEntryType) -> Void
+	) {
+		self.viewModel = viewModel
+		self.onAddEntryCellTap = onAddEntryCellTap
+
+		super.init(nibName: nil, bundle: nil)
+	}
+
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: - Overrides
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		navigationItem.largeTitleDisplayMode = .always
+		navigationItem.title = viewModel.day.formattedDate
+
+		view.backgroundColor = .enaColor(for: .darkBackground)
+
+		setupSegmentedControl()
+		setupTableView()
+
+		viewModel.$day
+			.receive(on: RunLoop.main)
+			.sink { [weak self] _ in
+				self?.updateForSelectedEntryType()
+			}
+			.store(in: &subscriptions)
+
+		viewModel.$selectedEntryType
+			.receive(on: RunLoop.main)
+			.sink { [weak self] _ in
+				// Scrolling to top prevents table view from flickering while reloading
+				self?.tableView.scrollToRow(at: IndexPath(row: 0, section: 0), at: .top, animated: false)
+				self?.updateForSelectedEntryType()
+			}
+			.store(in: &subscriptions)
+	}
+
+	// MARK: - Protocol UITableViewDataSource
+
+	func numberOfSections(in tableView: UITableView) -> Int {
+		return 2
+	}
+
+	func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		switch section {
+		case 0:
+			return 1
+		case 1:
+			return viewModel.entriesOfSelectedType.count
+		default:
+			fatalError("Invalid section")
+		}
+	}
+
+	func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		switch indexPath.section {
+		case 0:
+			return addCell(forRowAt: indexPath)
```

Comment:
It reads like a cell is added somewhere. 

```suggestion
			return makeAddCell(forRowAt: indexPath)
```

Reply:
This is from a different branch, the diff was not correct here in draft mode.